source/simulator/parameters.cc

```diff
@@ -1124,6 +1146,124 @@ namespace aspect
                      ExcMessage ("The number of multiple 'Global composition maximum' values "
                                  "and the number of multiple 'Global composition minimum' values "
                                  "have to be the same as the total number of compositional fields"));
+
+      compositional_field_methods
+        = Utilities::split_string_list
+          (prm.get ("Compositional field methods"));
+
+      AssertThrow ((compositional_field_methods.size() == 0) ||
+                   (compositional_field_methods.size() == 1) ||
+                   (compositional_field_methods.size() == n_compositional_fields),
+                   ExcMessage ("The length of the list of names for the field method of compositional "
+                               "fields needs to be empty, or have one entry, or have a length equal to "
+                               "the number of compositional fields."));
+
+      // If no method is specified set the default, which is solve every composition
+      // by a continuous field method
+      if (compositional_field_methods.size() == 0)
+        compositional_field_methods = std::vector<std::string> (n_compositional_fields,"continuous field");
```

Comment:
ah, so this would override the current way of selecting between continuous and discontinuous field by looking at the boolean flag that makes the decision for *all* compositional fields. Is your intent to be incompatible? It would make sense, since your approach allows making this decision field by field, rather than for *all* compositional fields at once.